packages/focusjail/src/useFocusJail.ts

```diff
@@ -10,11 +10,28 @@ import { composeEventHandlers, KEY_CODES } from '@zendeskgarden/container-utilit
 import tabbable from 'tabbable';
 import activeElement from 'dom-helpers/activeElement';
 
-export function useFocusJail({ focusOnMount = true, environment, focusElem, containerRef } = {}) {
+export interface IUseFocusJailProps {
+  focusOnMount?: boolean;
+  environment?: Document;
+  focusElem?: (element: HTMLElement) => any;
+  containerRef?: any;
```

Comment:
Would love some feedback here. 

I went with a type `any` here because `containerRef` is an optional argument, but we will manually throw a warning in `getContainerProps` if this argument isn't passed a ref. TypeScript is a bit unhappy when I type this value with `React.RefObject<any>`.

One alternative is to throw the error immediately in the hook, rather than in `getContainerProps` function. But that changes the API a little bit.

```jsx
export interface IUseFocusJailProps {
  focusOnMount?: boolean;
  environment?: Document;
  focusElem?: (element: HTMLElement) => any;
  containerRef?: React.RefObject<any>;
}

export const useFocusJail = ({
  focusOnMount = true,
  environment,
  focusElem,
  containerRef
}: IUseFocusJailProps = {}): IUseFocusJailReturnValue => {
  if (!containerRef) {
    throw new Error('Accessibility Error: You must apply the ref prop to your containing element.')
  }
  ...
}
```

Another alternative is that I can tell TypeScript that `containerRef` will always be available:

```jsx
export interface IUseFocusJailProps {
  focusOnMount?: boolean;
  environment?: Document;
  focusElem?: (element: HTMLElement) => any;
  containerRef?: React.RefObject<any>;
}

export const useFocusJail = ({
  focusOnMount = true,
  environment,
  focusElem,
  containerRef
}: IUseFocusJailProps = {}): IUseFocusJailReturnValue => {
  const [currentRef, setCurrentRef] = useState<HTMLElement>(containerRef!.current);

  // eslint-disable-next-line react-hooks/exhaustive-deps
  useEffect(() => {
    if (containerRef!.current !== currentRef) {
      setCurrentRef(containerRef!.current);
    }
  });
```

Reply:
> because containerRef is an optional argument

Is `containerRef` actually optional? It looks like it is required to be passed. I believe using `React.RefObject<HTMLElement>` could work here.

The `propTypes` of the container show it is `PropTypes.object.isRequired`

Reply:
Depending on how it's provided you may also need `React.RefObject<HTMLElement | undefined>`

Reply:
> Is containerRef actually optional? It looks like it is required to be passed. I believe using React.RefObject<HTMLElement> could work here.

Yes, it looks like it's required.

> I believe using React.RefObject<HTMLElement> could work here.

It works, however I needed to tell TypeScript to presume `containerRef` is available with the `!`. I pushed this up in a separate commit if you want to take a look.

Reply:
Update: `containerRef` is a required argument, and I have updated the interface reflect that by removing the optional `?` flag for `containerRef`.